### PR TITLE
Start removing `snippet_opt` in favor of `get_source_text`

### PR DESCRIPTION
**Pull request by Jarcho**
Continuing the job of removing unnecessary allocations.

changelog: none

### CODE REVIEW

File: clippy_lints/src/from_over_into.rs [hunk 1]
```diff
@@ -178,8 +178,10 @@ fn convert_to_from(
         return None;
     };
 
-    let from = snippet_opt(cx, self_ty.span)?;
-    let into = snippet_opt(cx, target_ty.span)?;
+    let from = self_ty.span.get_source_text(cx)?;
+    let into = target_ty.span.get_source_text(cx)?;
+    let from = from.as_str();
+    let into = into.as_str();
```
Comment (Alexendoo) [inline clippy_lints/src/from_over_into.rs:184]: These look like they'd be unneeded

File: clippy_lints/src/borrow_deref_ref.rs [hunk 2]
```diff
@@ -83,7 +84,7 @@ impl<'tcx> LateLintPass<'tcx> for BorrowDerefRef {
                     diag.span_suggestion(
                         e.span,
                         "if you would like to reborrow, try removing `&*`",
-                        snippet_opt(cx, deref_target.span).unwrap(),
+                        deref_text.as_str(),
```
[the remark this answers is not in the document]
Comment (Jarcho) [reply clippy_lints/src/borrow_deref_ref.rs:87]: `span_suggestion` only requires `ToString`. `deref_text` is used right afterwards so it doesn't help here.

[the remark this answers is not in the document]
Comment (Alexendoo) [reply clippy_lints/src/borrow_deref_ref.rs:87]: Ahh yeah, was thinking of the wrong arg 🤦‍♂️

**Review by Alexendoo — APPROVED** [review]

**Comment by Alexendoo** [comment]
Thanks!

@bors r+

**Pull request merged**